code/android-core-library/src/phone/java/com/adobe/marketing/mobile/MobileCore.java

```diff
@@ -250,118 +193,103 @@ public static void configureWithAppID(final String appId) {
      *
-     * @param event         required parameter, {@link Event} instance to be dispatched, should not be null
-     * @param errorCallback optional {@link ExtensionErrorCallback} which will be called if an error occurred during dispatching
-     * @return {@code boolean} indicating if the the event dispatching operation succeeded
+     * @param completionCallback An optional {@link AdobeCallback} invoked after registrations are completed
      */
-    public static boolean dispatchEvent(final Event event, final ExtensionErrorCallback<ExtensionError> errorCallback) {
-        if (eventHub == null) {
-            Log.debug(LOG_TAG, "Failed to dispatch event. (%s)", NULL_CONTEXT_MESSAGE);
+    public static void start(final AdobeCallback<?> completionCallback) {
+        if(!sdkInitializedWithContext.get()) {
+            Log.error(LOG_TAG, "Failed to start - completionCallback is null");
+            return;
+        }
 
-            if (errorCallback != null) {
-                errorCallback.error(ExtensionError.UNEXPECTED_ERROR);
-            }
+        EventHub.Companion.getShared().start();
 
-            return false;
+        if (completionCallback != null) {
+            completionCallback.call(null);
         }
 
-        if (event == null) {
-            Log.debug(LOG_TAG, "%s (Core.dispatchEvent) - The event was not dispatched", Log.UNEXPECTED_NULL_VALUE);
+    }
 
-            if (errorCallback != null) {
-                errorCallback.error(ExtensionError.EVENT_NULL);
-            }
+    /**
+     * Registers an event listener for the provided event type and source.
+     *
+     * @param eventType   required parameter, the event type as a valid string (not null or empty)
+     * @param eventSource required parameter, the event source as a valid string (not null or empty)
+     * @param callback    required parameter, {@link AdobeCallbackWithError#call(Object)} will be called when the event is heard
+     */
+    public static void registerEventListener(final String eventType, final String eventSource,
+                                             final AdobeCallbackWithError<Event> callback) {
+        if (callback == null) {
+            Log.error(LOG_TAG, "Failed to registerEventListener - callback is null",
+                    Log.UNEXPECTED_NULL_VALUE);
+            return;
+        }
 
-            return false;
+        if (eventType == null || eventSource == null) {
+            Log.error(LOG_TAG, "Failed to registerEventListener - event type/source is null");
+            callback.fail(AdobeError.UNEXPECTED_ERROR);
+            return;
         }
 
-        eventHub.dispatch(event);
-        return true;
+        EventHub.Companion.getShared().registerListener(eventType, eventSource, callback);
     }
 
     /**
      * This method will be used when the provided {@code Event} is used as a trigger and a response event
      * is expected in return. The returned event needs to be sent using
```

Comment:
Need to update javadocs here. `dispatchEvent` does not expect a response event in return right?